phobos/io/xml_factory.py

```diff
@@ -11,19 +11,15 @@
 FORMATS = json.load(open(pkg_resources.resource_filename("phobos", "data/xml_formats.json"), "r"))
 
 
-def is_int(val):
-    if type(val) == int:
-        return True
+def is_int(val: str):
     try:
         int(val)
         return True
     except ValueError:
         return False
 
 
-def is_float(val):
-    if type(val) == float:
-        return True
+def is_float(val: str):
```

Comment:
val is not necessarily string, This function shall check for any type. So this change has to be reverted.